tests/e2e/controlplane/control_plane_update_test.go

```diff
@@ -334,3 +334,41 @@ spec:
 		})
 	})
 })
+
+// getBaseAndNewVersion returns the base and new Istio versions to be used in the tests.
+// It will take the latest Major.Minor version and get the two newest patch of that Major.Minor version.
+func getBaseAndNewVersion() (string, string) {
+	type versionPair struct {
+		base, newVersion string
+		newSemver        *semver.Version
+	}
+	var latestPair versionPair
+
+	patchVersions := istioversion.GetPatchVersionsByMajorMinor()
+	for _, group := range patchVersions {
+		// Only consider groups with at least two patch versions.
+		if len(group) < 2 {
+			continue
+		}
+
+		newVer, err1 := semver.NewVersion(group[0].Name)
```

Comment:
`.Name` is the version name, not the semantic version number. You should probably use `.Version` here.